blocks/library/search/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+ import './editor.scss';
+import { registerBlockType } from '../../api';
+import InspectorControls from '../../inspector-controls';
+import TextControl from '../../inspector-controls/text-control';
+
+class SearchBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		this.updateLabel = this.updateLabel.bind( this );
+		this.updatePlaceholder = this.updatePlaceholder.bind( this );
+		this.updateSubmitValue = this.updateSubmitValue.bind( this );
+		this.handleFormSubmit = this.handleFormSubmit.bind( this );
+	}
+
+	updateLabel( newLabel ) {
+		this.props.setAttributes( { label: newLabel } );
+	}
+
+	updatePlaceholder( newPlaceholder ) {
+		this.props.setAttributes( { placeholder: newPlaceholder } );
+	}
+
+	updateSubmitValue( newValue ) {
+		this.props.setAttributes( { submitValue: newValue } );
+	}
+
+	handleFormSubmit( event ) {
+		event.preventDefault();
+		return false;
+	}
+
+	render() {
+		const {
+			attributes,
+			focus,
+		} = this.props;
+
+		const {
+			label,
+			placeholder,
+			submitValue,
+		} = attributes;
+
+		return [
+			<form
+				role="search"
+				method="get"
+				className="search-form"
+				onSubmit= { this.handleFormSubmit }
+			>
+				<label>
+					<span className="search-label">{ label }</span>
+					<input type="search" className="search-field" placeholder={ placeholder } value="" name="s" />
+				</label>
+				<input type="submit" className="search-submit" value={ submitValue } />
+			</form>,
+			focus && (
```

Comment:
`focus` was removed and we no longer need it.

`render` method might return `Fragment` component rather than array which needs to assign key to every child element. `form` doesn't have `key` provided so it will print warnings on the console.